op-service/sources/supervisor_client.go

```diff
@@ -174,6 +174,15 @@ func (cl *SupervisorClient) AllSafeDerivedAt(ctx context.Context, derivedFrom et
 	return result, err
 }
 
+func (cl *SupervisorClient) SyncStatus(ctx context.Context) (map[eth.ChainID]*eth.SupervisorChainStatus, error) {
+	var result map[eth.ChainID]*eth.SupervisorChainStatus
```

Comment:
The challenger doesn't actually want chain specific status here and I think it would be useful for supervisor's API to provide an overall sync status with the chain specific statuses included for reference (useful if you're debugging stuff or monitoring etc).

I suspect the only non-chain specific field would be the one thing challenger needs, which is the latest L1 block that has been fully processed. It's ok if that doesn't update immediately as long as it never reports an L1 block that hasn't been fully processed (ie it can lag behind a bit but it can't run ahead).

Reply:
I expect the challenger to check every returned chain status to determine the latest L1 head by itself. Seems nicer to be a bit more generic here by returning all chains statues for the new API than have it specific for the op-challenger.

Reply:
I just think it makes sense for op-supervisor to do that logic rather than every client having to.  That way you get the data if you manually call the sync endpoint as part of debugging  etc and don't have to scan the status of many different chains and work it out yourself.